pkg/vsphere/vm/vm.go

```diff
@@ -43,3 +49,63 @@ func NewVirtualMachine(ctx context.Context, session *session.Session, moref type
 		Session: session,
 	}
 }
+
+// FolderName returns the name of the namespace(vsan) or directory(vmfs) that holds the VM
+// this equates to the normal directory that contains the vmx file, stripped of any parent path
+func (vm *VirtualMachine) FolderName(ctx context.Context) (string, error) {
+	var mvm mo.VirtualMachine
+
+	if err := vm.Properties(ctx, vm.Reference(), []string{"runtime.host", "config"}, &mvm); err != nil {
+		log.Errorf("Unable to get managed config for VM: %s", err)
+		return "", err
+	}
+
+	path := path.Base(path.Dir(mvm.Config.Files.VmPathName))
```

Comment:
don't have the code to hand to check myself - make sure this path is the UUID based path on vSAN.